pkg/operator/audit/audit.go

```diff
@@ -0,0 +1,104 @@
+package audit
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	kyaml "k8s.io/apimachinery/pkg/util/yaml"
+	auditv1beta1 "k8s.io/apiserver/pkg/apis/audit/v1beta1"
+
+	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/v311_00_assets"
+	libgoapiserver "github.com/openshift/library-go/pkg/operator/configobserver/apiserver"
+)
+
+const (
+	policyAsset = "v3.11.0/openshift-apiserver/audit-policies-cm.yaml"
+)
+
+func NewAuditPolicyPathGetter() (libgoapiserver.AuditPolicyPathGetterFunc, error) {
+	policies, err := readPolicyNamesFromResource()
+	if err != nil {
+		return nil, err
+	}
+
+	return func(profile string) (string, error) {
+		const (
+			auditPolicyFilePath = "/etc/kubernetes/static-pod-resources/configmaps/openshift-apiserver-audit-policies"
+		)
+
+		// we expect the keys for audit profile in bindata to be in lower case and
+		// have a '.yaml' suffix.
+		key := fmt.Sprintf("%s.yaml", strings.ToLower(profile))
+		_, exists := policies[key]
+		if !exists {
+			return "", fmt.Errorf("invalid audit profile - key=%s", key)
+		}
+
+		return fmt.Sprintf("%s/%s", auditPolicyFilePath, key), nil
+	}, nil
+}
+
+// GetDefaultPolicy returns the default audit policy.
+func GetDefaultPolicy() (*auditv1beta1.Policy, error) {
```

Comment:
there is no bootstrap variant. Is this needed?

Reply:
removed